sklearn/model_selection/_validation.py

```diff
@@ -730,8 +730,9 @@ def cross_val_predict(estimator, X, y=None, *, groups=None, cv=None,
 
     method : str, default='predict'
         Invokes the passed method name of the passed estimator. For
-        method='predict_proba', the columns correspond to the classes
-        in sorted order.
+        methods {'decision_function', 'predict_proba', 'predict_log_proba'},
+        the columns correspond to the classes in order as sorted by
+        :class:`~sklearn.preprocessing.LabelEncoder`.
```

Comment:
is this the same as the order from the `classes_`? We could reference to the glossary entry.

In general this is the same order as calling any of these methods on the estimator, right?

Reply:
Yes, that is mostly the case and I was being nitpicky here. It's just that `cross_val_predict` does allow for the case that the training subset does not contain one or more classes, from doc:

> In the case that one or more classes are absent in a training portion, a default score needs to be assigned to all instances for that class if method produces columns per class, as in {‘decision_function’, ‘predict_proba’, ‘predict_log_proba’}. 

code:

https://github.com/scikit-learn/scikit-learn/blob/fd237278e895b42abe8d8d09105cbb82dc2cbba7/sklearn/model_selection/_validation.py#L752-L760

Reply:
I agree that the current

> For method='predict_proba', the columns correspond to the classes
        in sorted order

is not correct and we should remove it

But since this is about the shape of the output, I think we should mention that in the `Return` section instead.

Reply:
> I think we should mention that in the `Return` section instead.

Ah good point

> In general this is the same order as calling any of these methods on the estimator, right?

And I realise yes, when using all the data.

Do you think we should amend method to:
```
 method : {'predict', 'predict_proba', 'predict_log_proba', 'decision_function'}, default='predict'
```
I guess the methods could technically be anything, but I think these are the ones tested for?

Reply:
I amended the wording of 'method' as I thought it was hard to understand.